app/views/application/_usercard.html.slim

```diff
@@ -21,8 +21,8 @@ div class="avatarWithName #{'moderator' if user.is_moderator} #{'administrator'
         - user_dad_score = user.current_streak
         .levelOverlay.unselectable
           - if user_dad_score != 0
-            - next_level = BadgeMap.where("required_score > #{user_dad_score} AND challenge_id = 1").order("required_score ASC").first
-            - prev_level = BadgeMap.where("required_score <= #{user_dad_score} AND challenge_id = 1").order("required_score DESC").first
+            - next_level = BadgeMap.joins(:challenge).where("badge_maps.required_score > #{user_dad_score} AND challenges.end_date is null").order("required_score ASC").first
+            - prev_level = BadgeMap.joins(:challenge).where("badge_maps.required_score <= #{user_dad_score} AND challenges.end_date  is null").order("required_score DESC").first
```

Comment:
I would actually hold off on this, pending a discussion on whether or not there is ever a use case for an "unending challenge".
It's more likely that this will be something infrastructurally different (i.e. something like a "Draw a 10 page comic" achievement tied to some sort of tracking objective), but let's reopen this idea post challenge rework.
Do we have a better identifier for the DAD challenge? Name is flimsy, but it's all that comes to mind.

Reply:
it's mostly for dev purposes when i reset the database. i would recommend a special column in the db that indicates a challenge is special or something.
we really need to think about what to do with challenges. the whole thing needs restructuring.